lightning/src/ln/channelmanager.rs

```diff
@@ -4110,6 +4157,24 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			return Err(MsgHandleErrInternal::send_err_msg_no_close("No inbound channels accepted".to_owned(), msg.temporary_channel_id.clone()));
 		}
 
+		if self.default_configuration.manually_accept_inbound_channels {
+			let mut pending_events = self.pending_events.lock().unwrap();
+			pending_events.push(
+				events::Event::OpenChannelRequest {
+					open_channel_request_info: OpenChannelRequestInfo {
+						counterparty_node_id: counterparty_node_id.clone(),
+						their_features: their_features.clone(),
+						msg: msg.clone()
+					},
+				}
+			);
+			return Ok(())
+		}
+
+		return self.internal_create_requested_channel(counterparty_node_id, their_features, msg);
+	}
+
+	fn internal_create_requested_channel(&self, counterparty_node_id: &PublicKey, their_features: InitFeatures, msg: &msgs::OpenChannel) -> Result<(), MsgHandleErrInternal>{
```

Comment:
Instead of splitting the function here, we can simplify things a bit by splitting the function a few lines down and moving only the `SendAcceptChannel` message generation out. There's no harm in having a channel be inserted into the `channel_state.by_id` map even if its not accepted yet, we just have to make sure we reject a `FundingCreated` message in `Channel::funding_created` if the sender is trying to be too clever and skip waiting for our accept. This will require a new flag in the `Channel` which is updated when we (eventually) call `get_accept_channel` and checked in `funding_created`.

From there, `Event::OpenChannelRequest` can be tweaked to just include relevant info instead of doing it via a struct, and `respond_to_open_channel_request` can just take the `temporary_channel_id`. All that said, I'd suggest we err on the side of including too much data rather than too little, as the node operator may need extra data to decide if they want to accept a channel.

Reply:
Thanks a lot for the detailed input! I'll start working on implementing your suggestions. 

Just to clarify though, how much/which data is enough for node operator to aid them in the decision to accept or reject the channel? Including all fields in the `OpenChannel` msg is too excessive, but given that I'd really appreciate if I'd could get some feedback on which fields should be included instead of selecting myself.

Reply:
I'm not sure there's necessarily a clear answer, but maybe include, for now (we can always add more later): counterparty node id, chain_hash, temporary_channel_id, funding_satoshis, push_msat?

Reply:
Thanks, I'll start with those.